### PR TITLE
Add StopForumSpam extension

### PR DESCRIPTION
**Pull request by amrav**
This extension checks edits against an updated list of spammer IPs across various forums, and prevents edits and account creations from those IPs.

Set up nightly cron job to update spam blacklist.

Ran locally and confirmed that adding my local IP to the blacklist prevented me from editing. Also confirmed that the update cron job works as expected.

### CODE REVIEW

File: parsoid/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ WORKDIR /home/parsoid
 
 RUN git clone https://gerrit.wikimedia.org/r/p/mediawiki/services/parsoid && \
     cd parsoid && \
-    git checkout v0.8.1 && \
+    git checkout v0.10.0 && \
```
Comment (amrav) [inline parsoid/Dockerfile:11]: Took the opportunity to bump node and parsoid, but this is unrelated and shouldn't make a difference.

File: jobs/crontab [hunk 2]
```diff
@@ -2,3 +2,4 @@
 
 11 * * * * /root/update_top_trending.sh
 * * * * 1 /root/MetaMaint.sh
+47 3 * * * /root/update_spam_blacklist.sh
```
Comment (icyflame) [inline jobs/crontab:5]: Any specific reason for this time? If there is none, I think we should change it to something less specific, `00:01` or `01:00` sound like good times to me.

**Review by icyflame — APPROVED** [review]
LGTM. Cron time is strange, but it's fine even if left unchanged.

**Comment by amrav** [comment]
Haha, I read somewhere that it's good to set cron times to something random
in the range you want, to avoid many expensive jobs running at the same
Schelling point (eg on the hour). Not really an issue for us but seems like
good practice.

Anecdotal aside: I've seen a system exceed capacity because some job
scheduling web UI defaulted to midnight, and most users just went with the
default.
